### ISSUE TITLE
It seems `Fsa.Invert_` does not swap `labels` and `aux_labels`?

### ISSUE THREAD

**Issue by qindazhu** [k2-fsa/k2#323]
@csukuangfj , not sure if I used `Invert_` correctly, but here is the example I tested with (copying from you colob scripts):
```
 L_str = '''
      0 0 1 0
      0 0 1 0
      0 0 2 0
      0 0 2 0
      0 0 3 0
      0 0 3 0
      0 1 -1 0
      1
      '''
  L = k2.Fsa.from_str(L_str)
  L.aux_labels = torch.tensor([1, 2, 1, 2, 1, 2, -1]).to(torch.int32)

  print(L.aux_labels)
  print(L.labels)
  # L = k2.arc_sort(L.invert_())
  L.invert_()
  print(L.aux_labels)
  print(L.labels)


```

The output is
```
tensor([ 1,  2,  1,  2,  1,  2, -1], dtype=torch.int32)
tensor([ 1,  1,  2,  2,  3,  3, -1], dtype=torch.int32)
tensor([ 1,  2,  1,  2,  1,  2, -1], dtype=torch.int32)
tensor([ 1,  2,  1,  2,  1,  2, -1], dtype=torch.int32)
```
It seems we did not swap `labels` and `aux_labels`?

**Comment by csukuangfj** [comment]
Please pull the latest master. It has been fixed.

**Comment by qindazhu** [comment]
OK, thanks

**Comment by qindazhu** [comment]
Is there any extra step I need to do when building pip package, I built and installed successfully (following build_pip.sh), but when I run with `k2.Fsa.from_str(str)`, would get segmentation fault.

```
#1  0x00007fff91e3ee58 in void k2::Eval<__nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (k2::Array1<int>::*)(int), &k2::Array1<int>::operator=, 1u>, void (int), int*, int const> >(CUstream_st*, int, __nv_hdl_wrapper_t<false, false, __nv_dl_tag<void (k2::Array1<int>::*)(int), &k2::Array1<int>::operator=, 1u>, void (int), int*, int const>&) () from /ceph-hw/.local/lib/python3.8/site-packages/libcontext.so
#2  0x00007fff91e871dc in k2::FsaFromArray1(k2::Array1<k2::Arc>&, bool*) () from /ceph-hw/.local/lib/python3.8/site-packages/libcontext.so
#3  0x00007fff91eb41c4 in k2::FsaFromString(std::string const&, bool, k2::Array1<int>*) () from /ceph-hw/.local/lib/python3.8/site-packages/libcontext.so
#4  0x00007fffa56894f8 in ?? () from /ceph-hw/.local/lib/python3.8/site-packages/_k2.cpython-38-x86_64-linux-gnu.so
#5  0x00007fffa56786a9 in ?? () from /ceph-hw/.local/lib/python3.8/site-packages/_k2.cpython-38-x86_64-linux-gnu.so
#6  0x00007ffff792447c in cfunction_call_varargs () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#7  0x00007ffff7924a31 in _PyObject_MakeTpCall () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#8  0x00007ffff7901213 in _PyEval_EvalFrameDefault () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#9  0x00007ffff79deb04 in _PyEval_EvalCodeWithName () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#10 0x00007ffff7924828 in _PyFunction_Vectorcall () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#11 0x00007ffff7927e10 in method_vectorcall () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#12 0x00007ffff7902c26 in _PyEval_EvalFrameDefault () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#13 0x00007ffff79deb04 in _PyEval_EvalCodeWithName () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#14 0x00007ffff79ded0e in PyEval_EvalCodeEx () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#15 0x00007ffff79ded3b in PyEval_EvalCode () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#16 0x00007ffff7a144a8 in run_eval_code_obj () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#17 0x00007ffff7a14560 in run_mod () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#18 0x00007ffff7a16e45 in PyRun_FileExFlags () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#19 0x00007ffff7a16fb6 in PyRun_SimpleFileExFlags () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#20 0x00007ffff7a31708 in Py_RunMain () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#21 0x00007ffff7a31b39 in Py_BytesMain () from /home/linuxbrew/.linuxbrew/Cellar/python@3.8/3.8.6_1/lib/libpython3.8.so.1.0
#22 0x00007ffff72a0b97 in __libc_start_main (main=0x4008a0 <main>, argc=2, argv=0x7fffffffe438, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffe428)

```

However it's fine when I use the package from pypi, system related issue?

**Comment by danpovey** [comment]
Do the tests run?  And you're sure that's where it installed it?

**Comment by csukuangfj** [comment]
Is the GPU you are using not used by anyone else?

**Comment by qindazhu** [comment]
There's no any GPU usage on my machine. 


I just followed the scripts here https://github.com/k2-fsa/k2/blob/master/scripts/build_pip.sh to build and then run

`pip install dist/k2_some_suffix` to install, It can be imported correctly but just failed when I call `k2.Fsa.from_str()`

**Comment by csukuangfj** [comment]
All steps are documented in `build_pip.sh`. Is the log you posted above complete?

**Comment by qindazhu** [comment]
Yeah, noted that I got it with `backtrace` in `gdb`. The program just fail with segmentation fault

```
>>> import k2
>>> a = '''
... 0 0 1 0
... 0 1 -1 0
... 1
... '''
>>> a
'\n0 0 1 0\n0 1 -1 0\n1\n'
>>> b = k2.Fsa.from_str(a)
[1]    1363043 segmentation fault  python
```

**Comment by csukuangfj** [comment]
How about `import torch` first and then `import k2`?

**Comment by qindazhu** [comment]
The same error

```
Python 3.8.6 (default, Oct 29 2020, 22:20:58)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import k2
>>> a = '''
... 0 0 1 0
... 0 1 -1 0
... 1
... '''
>>> a
'\n0 0 1 0\n0 1 -1 0\n1\n'
>>> b = k2.Fsa.from_str(a)
[1]    1367540 segmentation fault  python

```

Maybe it's better if you can merge your PR about building pip package with Github actions, so I can just download the package on github and try with it (not sure if it works). Let's not take too much time on this  for now as I'm wrinting the training scripts.

**Comment by csukuangfj** [comment]
I am using google drive to host the files of cudnn. It seems that I have downloaded them multiple times within 24 hours
and Google won't allow GitHub actions to download them today. I am finding a new place to host the files.

**Comment by qindazhu** [comment]
I think @danpovey may have some servers to host those files?

**Comment by csukuangfj** [comment]
PyPI contains a package for python 3.8 built against the latest master branch.

**Comment by qindazhu** [comment]
Many thanks!

**Comment by danpovey** [comment]
I managed to get this to work myself:

de-74279-k2-dev-0929153814-7686b86bc5-qbfjq:~: python3
Python 3.8.6 (default, Oct 29 2020, 22:20:58)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import k2
>>> a='\n0 0 1 0\n0 1 -1 0\n1\n'                                                                                                                                                          

>>> >>> k2.Fsa.from_str(a)



<k2.fsa.Fsa object at 0x7fd888613e20>
>>> >>> >>> >>>

**Comment by danpovey** [comment]
You might be able to get more information about this by doing it as:
 gdb python
(gdb) r
...
and then the rest that you did..
it might at least give you a stack trace.
